src/main/java/codeu/model/data/User.java

```diff
@@ -14,61 +14,109 @@
-     *
-     * @param id           the ID of this User
-     * @param name         the username of this User
-     * @param passwordHash the password hash of this User
-     * @param creation     the creation time of this User
-     */
-    public User(UUID id, String name, String passwordHash, Instant creation) {
-        this.id = id;
-        this.name = name;
-        this.passwordHash = passwordHash;
-        this.creation = creation;
-    }
-
-    /**
-     * Returns the ID of this User.
-     */
-    public UUID getId() {
-        return id;
-    }
-
-    /**
-     * Returns the username of this User.
-     */
-    public String getName() {
-        return name;
-    }
-
-    /**
-     * Returns the password hash of this User.
-     */
-    public String getPasswordHash() {
-        return passwordHash;
-    }
-
-    /**
-     * Returns the creation time of this User.
-     */
-    public Instant getCreationTime() {
-        return creation;
-    }
 
+  private final UUID id;
+  private final String name;
+  private final String passwordHash;
+  private final Instant creation;
+  private Set<String> friends;
+  private UserStore userStore;
+
+  /**
+   * Constructs a new User.
+   *
+   * @param id the ID of this User
+   * @param name the username of this User
+   * @param passwordHash the password hash of this User
+   * @param creation the creation time of this User
+   * @param friends a set with the usernames of the users friends
+   */
+  public User(UUID id, String name, String passwordHash, Instant creation, Set<String> friends) {
+    this.id = id;
+    this.name = name;
+    this.passwordHash = passwordHash;
+    this.creation = creation;
+    this.friends = friends;
+    userStore = UserStore.getInstance();
+  }
+
+  /**
+   * Constructs a new User.
+   *
+   * @param id the ID of this User
+   * @param name the username of this User
+   * @param passwordHash the password hash of this User
+   * @param creation the creation time of this User
+   */
+  public User(UUID id, String name, String passwordHash, Instant creation) {
+    this(id, name, passwordHash, creation, new HashSet<>());
+  }
+
+  /** Returns the ID of this User. */
+  public UUID getId() {
+    return id;
+  }
+
+  /** Returns the username of this User. */
+  public String getName() {
+    return name;
+  }
+  
+  /** Returns the password hash of this User. */
+  public String getPasswordHash() {
+    return passwordHash;
+  }
+
+  /** Returns the creation time of this User. */
+  public Instant getCreationTime() {
+    return creation;
+  }
+
+  /** Returns a set with the usernames of this User's friends. */
+  public Set<String> getFriends() {
+    return friends;
+  }
+
+  /**
+   * Adds friend with given id if User with given id exists
+   *
+   * @param username the username of the friend to be added
+   */
+  public void addFriend(String username) {
+      if (userStore.getUser(id) != null) {
+        friends.add(username);
+      }
+  }
```

Comment:
I'm a bit confused here- what happens if a user tries adding a new friend that was already added? Additionally, where is the parameter id coming from, wouldn't you need to call getID() to retrieve it?

Reply:
You can get a user by id. Id is a parameter that. Adding a repeated friend will do nothing du to how Sets work. Sets do not allow repeats. You can take a look at the Java docs if you are interested https://docs.oracle.com/javase/7/docs/api/java/util/Set.html